src/factors/Pose2D.jl

```diff
@@ -42,17 +44,41 @@ end
 Pose2Pose2(::UniformScaling) = Pose2Pose2() # MvNormal(zeros(3),LinearAlgebra.diagm([1.0;1.0;1.0])) )
 
 
-getSample(cf::CalcFactor{<:Pose2Pose2}, N::Int=1) = (rand(cf.factor.z,N), )
+function getSample(cf::CalcFactor{<:Pose2Pose2}, N::Int=1) 
+  # Z = cf.factor.Z
+  # M = getManifold(Pose2)
+  # p = getPointIdentity(Pose2)
+  
+  # Xc = [rand(Z) for _ in 1:N]
+  
+  # # X = get_vector.(Ref(M), Ref(p), Xc, Ref(DefaultOrthogonalBasis()))
+  # X = hat.(Ref(M), Ref(p), Xc)
+  # points = exp.(Ref(M), Ref(p), X)
+
+  # (points, )
+
+  Xc = [rand(cf.factor.z) for _ in 1:N]
+  (Xc, )
+end
+
+
+# function (cf::CalcFactor{<:Pose2Pose2})(meas,
+#                                         wxi,
+#                                         wxj  )
+#   #
+
+#     #
+#     wTjhat = SE2(wxi)*SE2(meas)
+#     jTjhat = SE2(wxj) \ wTjhat
+#     return se2vee(jTjhat)
+# end
 
-function (cf::CalcFactor{<:Pose2Pose2})(meas,
-                                        wxi,
-                                        wxj  )
-  #
+function (cf::CalcFactor{<:Pose2Pose2})(Xc, p, q)
+    M = getManifold(cf.factor)
+    X = hat(M, p, Xc)
 
-    #
-    wTjhat = SE2(wxi)*SE2(meas)
-    jTjhat = SE2(wxj) \ wTjhat
-    return se2vee(jTjhat)
+    q̂ = Manifolds.compose(M, p, exp(M, identity(M, p), X)) #for groups
+    return log(M, q, q̂)
```

Comment:
Note, AbstractManifoldMinimize currently expects residual to be a vector in the tangent space